### PR TITLE
Update Catppuccin extension

### PR DESCRIPTION
**Pull request by uncenter**
## Description

Updates the README text to match Catppuccin website prose, fixes a link to the docs. Various improvements to the search palette command: removes the inset around grid items / colors so that colors fill the entire item, displays color names as grid item titles instead of slugs/identifiers, and adds copy name and copy identifier actions.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/catppuccin/metadata/catppuccin-5.png
Comment (uncenter) [inline extensions/catppuccin/metadata/catppuccin-5.png:1]: Doesn't look like the icon colors are Catppuccin-ized in these preview screenshots.

File: extensions/catppuccin/metadata/catppuccin-6.png
Comment (uncenter) [inline extensions/catppuccin/metadata/catppuccin-6.png:1]: Same here about the icon colors.

File: extensions/catppuccin/package.json [hunk 1]
```diff
@@ -85,6 +85,32 @@
...
+        {
+          "title": "Mocha",
+          "value": "mocha"
+        },
+        {
+          "title": "Macchiato",
+          "value": "macchiato"
+        },
+        {
+          "title": "Frappe",
```
Comment (uncenter) [inline extensions/catppuccin/package.json:106]:
```suggestion
          "title": "Frappé",
```
This should technically be the name of frappe if Raycast handles the accents fine.

File: extensions/catppuccin/src/search-ports.tsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-
-interface PortDetails {
-  name: string;
-  platform?: string | string[];
-  categories?: string[];
-  icon?: string;
-  color?: string;
-}
+import { getDefaultIcon, getIcon } from "./utils/icons.util";
+import { PortDetails } from "./types";
```
Comment (uncenter) [inline extensions/catppuccin/src/search-ports.tsx:4]:
```suggestion
import type { PortDetails } from "./types";
```

File: extensions/catppuccin/src/search-userstyles.tsx [hunk 3]
```diff
@@ -1,19 +1,8 @@
...
-  color?: string;
-  readme?: {
-    "app-link"?: string | string[];
-    usage?: string;
-  };
-  currentMaintainers?: string[];
-  pastMaintainers?: string[];
-}
+import { getDefaultIcon, getIcon } from "./utils/icons.util";
+import { UserStyleDetails } from "./types";
```
Comment (uncenter) [inline extensions/catppuccin/src/search-userstyles.tsx:5]:
```suggestion
import type { UserStyleDetails } from "./types";
```

File: extensions/catppuccin/src/utils/icons.util.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    return getDefaultIcon(color);
+  }
+
+  const normalizedIconName = iconName.toLowerCase();
+  const formattedColor = formatColor(color);
+
+  // check if icon ends .svg indiciating that it must be grabbed from the website repo
+  if (normalizedIconName.endsWith(".svg")) {
+    return {
+      source: `https://raw.githubusercontent.com/catppuccin/website/refs/heads/main/src/icons/ports/${normalizedIconName}`,
```
Comment (uncenter) [inline extensions/catppuccin/src/utils/icons.util.ts:42]: The ref should be pinned to a hash here right?

File: extensions/catppuccin/src/utils/palette.util.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import type { CatppuccinFlavor } from "@catppuccin/palette";
+import { flavors } from "@catppuccin/palette";
+
+export type CatppuccinFlavors = typeof flavors;
+export type FlavorName = keyof CatppuccinFlavors;
```
Comment (uncenter) [inline extensions/catppuccin/src/utils/palette.util.ts:5]: Both of these are already types provided by `@catppuccin/palette` iirc.

File: extensions/catppuccin/metadata/catppuccin-5.png
Comment (dvhsh) [reply extensions/catppuccin/metadata/catppuccin-5.png:1]: updated screenshot to use mocha flavor

File: extensions/catppuccin/metadata/catppuccin-6.png
Comment (dvhsh) [reply extensions/catppuccin/metadata/catppuccin-6.png:1]: updated screenshot to use mocha flavor

File: extensions/catppuccin/package.json (hunk 1 above)
Comment (dvhsh) [reply extensions/catppuccin/package.json:106]: raycast handles the accents no issue, updated

File: extensions/catppuccin/src/search-ports.tsx (hunk 2 above)
Comment (dvhsh) [reply extensions/catppuccin/src/search-ports.tsx:4]: fixed

File: extensions/catppuccin/src/search-userstyles.tsx (hunk 3 above)
Comment (dvhsh) [reply extensions/catppuccin/src/search-userstyles.tsx:5]: fixed

**Comment by dvhsh** [comment]
I'd like to leave it as a note here that when using icons of a certain color (tintColor), raycast itself appears to render certain colors incorrectly? i.e #d20f39 shows as #D10D39

File: extensions/catppuccin/src/utils/icons.util.ts (hunk 4 above)
Comment (dvhsh) [reply extensions/catppuccin/src/utils/icons.util.ts:42]: now using  69e2f3ee385279ea34e84c2c42703ed997eab40c

File: extensions/catppuccin/src/utils/palette.util.ts (hunk 5 above)
Comment (dvhsh) [reply extensions/catppuccin/src/utils/palette.util.ts]: now using FlavorName from `@catppuccin/palette`

**Comment by dvhsh** [comment]
> tintColor has [adjustContrast](https://developers.raycast.com/api-reference/user-interface/colors#properties), that should help you preserve the colors you're after

Ah I see now, I'll adjust this real quick

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**